Teensy 4.0 and 4.1 support

ccrome opened the pull request:
**NOTE:  I don't know the appropriate VID/PID -- It's currently using the 1060_evk settings and not the teensy settings.  Not sure which to use**

## Checklist

- [x] Please provide specific title of the PR describing the change
- [x] Please provide related links (*eg. Issue which will be closed by this Pull Request*).
- [x] If you are adding an new boards, please make sure
  - [x] Provide link to your allocated VID/PID if applicable.  **I don't know what do do here.  Currently, I left it at the values for the 1060_evk**
  - [x] Add your board to [action ci](/.github/workflows) in correct workflow and alphabet order for release binary
  - [x] `UF2_BOARD_ID` in your board.h follow correct format from [uf2 specs](https://github.com/microsoft/uf2#files-exposed-by-bootloaders)

-----------

## Description of Change
This commit adds support for the Teensy 4.0 and 4.1 boards.

It modifies the ports/mimxrt10xx/tusb_config.h and boards.c so that either USB port 0 or usb port 1 can be used.  This is useful for any IMXRT1062 based board.  Though less useful for the teensy 4.0 because it doens't have that port pinned out.

The discussion of this PR is here: https://github.com/adafruit/tinyuf2/discussions/285

hathach commented on ports/mimxrt10xx/README.md:
```diff
@@ -49,6 +49,9 @@ Double tap to enter bootloader mode, then simply drag & drop `update-tinyuf2_BOA
 - [MIMX RT1010 Evaluation Kit](https://www.nxp.com/design/development-boards/i.mx-evaluation-and-development-boards/i.mx-rt1010-evaluation-kit:MIMXRT1010-EVK)
 - [MIMX RT1020 Evaluation Kit](https://www.nxp.com/design/development-boards/i.mx-evaluation-and-development-boards/i.mx-rt1020-evaluation-kit:MIMXRT1020-EVK)
 - [MIMX RT1060 Evaluation Kit](https://www.nxp.com/design/development-boards/i.mx-evaluation-and-development-boards/mimxrt1060-evk-i.mx-rt1060-evaluation-kit:MIMXRT1060-EVK)
+- [Teensy 4.0](boards/teensy40)
```
must linked to manufacturer url

ccrome replied:
> thank you for the PR, please check out review for suggested changes. Also what is the use of **nxp_blhost_sdphost** submodule ? if not needed, we should removed that

Oops, accidentally left that in.  Removed via 6031ec2ddd5b4db11dd9a5c7b8bb3aa7699b3673

hathach commented on ports/mimxrt10xx/port.mk:
```diff
@@ -15,7 +15,11 @@ CFLAGS += \
-  
+
+# Choose which port to use
+TUD_OPT_RHPORT = 0
```
`TUD_OPT_RHPORT` is defined internally by tinyusb, this must be renamed to other such as BOARD_TUD_RHPORT

ccrome replied:
fixed via 9db2504e0443c3b80858c4e6203ef4e05cc2f2de

hathach commented on ports/mimxrt10xx/boards/teensy40/board.h:
```diff
@@ -0,0 +1,82 @@
+//--------------------------------------------------------------------+
+
+#define USB_VID           0x239A
+#define USB_PID           0x0083
```
adafruit does allocated an ID for teensy 4.0 which is `0x0085`

ccrome replied:
fixed via 9db2504e0443c3b80858c4e6203ef4e05cc2f2de

hathach commented on ports/mimxrt10xx/boards/teensy41/board.h:
```diff
@@ -0,0 +1,82 @@
+//--------------------------------------------------------------------+
+
+#define USB_VID           0x239A
+#define USB_PID           0x0083
```
must be changed to `0x00AD`

ccrome replied:
fixed via 9db2504e0443c3b80858c4e6203ef4e05cc2f2de

hathach commented on ports/mimxrt10xx/boards/teensy41/board.h:
```diff
@@ -0,0 +1,82 @@
+#define USB_VID           0x239A
+#define USB_PID           0x0083
+#define USB_MANUFACTURER  "PJRC"
+#define USB_PRODUCT       "TEENSY-4.1"
```
should be `Teensy 4.1`

hathach commented on ports/mimxrt10xx/boards/teensy40/board.h:
```diff
@@ -0,0 +1,82 @@
+#define USB_VID           0x239A
+#define USB_PID           0x0083
+#define USB_MANUFACTURER  "PJRC"
+#define USB_PRODUCT       "TEENSY-4.0"
```
should be `Teensy 4.0`

ccrome replied:
fixed via 9db2504e0443c3b80858c4e6203ef4e05cc2f2de

hathach commented on ports/mimxrt10xx/boards/teensy40/board.h:
```diff
@@ -0,0 +1,82 @@
+
+#define UF2_PRODUCT_NAME  USB_MANUFACTURER " " USB_PRODUCT
+#define UF2_BOARD_ID      "MIMXRT1060-TEENSY-40"
+#define UF2_VOLUME_LABEL  "TEENSY40BT"
```
should have BOOT at the end, maybe "TNSY40BOOT", applies to 4.1 as well.

hathach reviewed (changes requested):
thank you for the PR, please check out review for suggested changes. Also what is the use of **nxp_blhost_sdphost** submodule ? if not needed, we should removed that

ccrome commented:
I believe I addressed all the issues with 9db2504e0443c3b80858c4e6203ef4e05cc2f2de.  I ran and verified that things still work on teensy 4.1 and 4.0, and that the PIDs are correct, and that the boot disks show up correctly.

hathach reviewed (approved):
thank you very much for the PR, I made some additional changes mostly for adding hex rule for imxrt. It shouldn't change the behavior of this PR. Will merge when ci passed

203Null commented:
Anyway to convert a hex to uf2?
